src/digest/sha1.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2015-2016 Brian Smith.
+// Copyright 2016 Simon Sapin.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use core;
+use core::num::Wrapping;
+use c;
+use polyfill;
+use super::{Algorithm, sha256_format_output, MAX_CHAINING_LEN};
+
+/// SHA-1 as specified in [FIPS
+/// 180-4](http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf).
+///
+/// SHA-1 is deprecated in *ring*.
+/// The main purpose in retaining it is to support legacy protocols and OCSP,
+/// none of which need a fast SHA-1 implementation.
+/// This implementation therfore favors size and simplicity over speed.
+/// Unlike SHA-256, SHA-384, and SHA-512, there is no assembly language implementation.
```

Comment:
Please move the text "The main purpose...assembly language implementation." to be directly above the code implementing the algorithm. In particular, it doesn't make sense for this text to appear in the rustdoc for the module. Also, people are more likely to be wondering about this design choice when reading that code than when reading the definition of the `SHA1` static.